### PR TITLE
ATO-561: Enforce client registry permitted levels of confidence

### PR DESCRIPTION
**Pull request by hannah-leong**
## What

Add checks to the validation to make sure that requested LoCs are permitted in the client registry for the given client

## Checklist

- [ ] Impact on orch and auth mutual dependencies has been checked.

- [ ] A UCD review has been performed.

### CODE REVIEW

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/validators/QueryParamsAuthorizeValidator.java [hunk 1]
```diff
@@ -135,6 +147,17 @@ private boolean areScopesValid(List<String> scopes, ClientRegistry clientRegistr
         return true;
     }
 
+    private boolean areLevelOfConfidenceValuesPermitted(
```
Comment (ethanmills) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/validators/QueryParamsAuthorizeValidator.java:150]: Is this used anywhere?

Comment (hannah-leong) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/validators/QueryParamsAuthorizeValidator.java:150]: ... 😳

**Review by ethanmills — DISMISSED** [review]
Let's deploy this one carefully

**Review by ethanmills — APPROVED** [review]
Can we test this in staging please?

**Pull request merged**